### PR TITLE
add pc.route decorator

### LINKED ISSUE #462: Add wrapper for adding a page
Adding a function wrapper for adding a page is more pythonic and would be cleaner instead of dumping all your add_page's at the bottom.
Also don't remove the actual add_page function, just add an alternate wrapper method.

### PR DESCRIPTION
**Pull request by Lendemor**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

You can now decorate any function that return a Pynecone Component with `@pc.route`

`pc.route` accept the same argument as `add_page`.

Decorated functions still need to at least be imported in your main pynecone file.

Closes #462

### CODE REVIEW

File: pynecone/route.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+def route(route=None, title=None, image=None, description=None, on_load=None):
+    """Decorate a function as a page.
+
+    pc.App() will automatically call add_page() for any method decorated with route
+    when App.compile is called.
+
+    Note: the decorated functions still need to be imported
+
+    Args:
+        route (_type_, optional): The route to reach the page. Defaults to None.
```
Comment (picklelo) [inline pynecone/route.py:15]: Let's add the type annotations in the function signature and remove them from the docstrings

**Review by picklelo — COMMENTED** [review]
Awesome! Just a small nit comment

File: pynecone/route.py (hunk 1 above)
Comment (Lendemor) [reply pynecone/route.py:15]: sure

**Review by picklelo — APPROVED** [review]
Awesome

**Pull request merged**